### PR TITLE
Options, Meta APIs: Don't save invalid timezone values.

### PR DESCRIPTION
**Pull request by costdev**
Previously, timezone values such as an empty string, 0 or an invalid timezone identifier were saved.
This produced a warning in PHP < 7.4, and a Fatal Error in PHP 8+.

This adds validation which resets invalid values to the previous value before saving, and displays an error message to the user.

Includes E2E tests.

Trac ticket: https://core.trac.wordpress.org/ticket/58814

### CODE REVIEW

**Comment by costdev** [comment]
Note: E2E and Performance workflow failures are due to a known timeout issue. The tests added in this PR are passing.

File: src/wp-admin/options.php [hunk 1]
```diff
@@ -279,6 +279,23 @@
 			$_POST['gmt_offset']      = $_POST['timezone_string'];
 			$_POST['gmt_offset']      = preg_replace( '/UTC\+?/', '', $_POST['gmt_offset'] );
 			$_POST['timezone_string'] = '';
+		} elseif ( isset( $_POST['timezone_string'] ) && ! in_array( $_POST['timezone_string'], timezone_identifiers_list(), true ) ) {
```
Comment (peterwilsoncc) [inline src/wp-admin/options.php]: Mayybe?

I'm thinking of covering cases in which timezones have changed name, such as Kiev/Kyiv.

```suggestion
		} elseif ( isset( $_POST['timezone_string'] ) && ! in_array( $_POST['timezone_string'], timezone_identifiers_list( DateTimeZone::ALL_WITH_BC ), true ) ) {
```

File: tests/e2e/specs/general-settings-invalid-timezone.test.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
+import {
```
Comment (peterwilsoncc) [inline tests/e2e/specs/general-settings-invalid-timezone.test.js]: Three days ago the e2e tests were migrated to playwright 5a838d1bb7e6e1f5aa5f7f8e9924cb8fb67982df

File: src/wp-admin/options.php (hunk 1 above)
Comment (costdev) [reply src/wp-admin/options.php]: Great point! The form on `options-general.php` may have an earlier, now outdated timezone string added to the options list per [this commit](https://github.com/WordPress/wordpress-develop/commit/45da79e52e3252ddb8286ca0e6cd5db4522d874ca), where the `value` attribute is set to the same outdated value, not the potential "new" value. That means `$_POST['timezone_string']` _can_ be an outdated value, and would therefore be missed by this code. Using the `ALL_WITH_BC` constant isn't just a good idea, it's _required_ for backward compatibility.

Very nice catch Peter!

File: tests/e2e/specs/general-settings-invalid-timezone.test.js (hunk 2 above)
Comment (costdev) [reply tests/e2e/specs/general-settings-invalid-timezone.test.js]: Tests converted to Playwright in [2b952d6](https://github.com/WordPress/wordpress-develop/pull/4853/commits/2b952d69762b884cc570c074453c23fbcf6ccc68).

**Review by peterwilsoncc — APPROVED** [review]
LGTM, I can commit this during my day.

**Comment by peterwilsoncc** [comment]
Committed [r56949](https://core.trac.wordpress.org/changeset/56949) / https://github.com/WordPress/wordpress-develop/commit/c626e637154f175806c4d4bf252c01ec9746a73e

**Pull request closed without merging**